PayCalculatorAPI/Controllers/EmployeeController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Employees.Entites;
+using Employees.Entities.JSON;
+using Microsoft.AspNetCore.Mvc;
+using NPOI.SS.Formula.Functions;
+using System.Collections.Generic;
+
+namespace PayCalculatorAPI.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class EmployeeController : ControllerBase
+    {
+        private readonly EmployeeJSONFormatter<PermanentEmployee> _permanentJSONFormatter;
+        private readonly EmployeeJSONFormatter<TempEmployee> _tempJSONFormatter;
+        public EmployeeController(EmployeeJSONFormatter<PermanentEmployee> permanentJSONFormatter, EmployeeJSONFormatter<TempEmployee> tempJSONFormatter)
+        {
+            _permanentJSONFormatter = permanentJSONFormatter;
+            _tempJSONFormatter = tempJSONFormatter;
+        }
+        [HttpGet]
+        public ActionResult<List<T>> Get(EmployeeType employeeType)
+        {
+            var permanentEmployees = _permanentJSONFormatter.ReadFromList();
```

Comment:
You could make this more efficient by not calling the `ReadFromList()` methods until you know what you need. 

It won't make a visible difference with just a handful of employees, but imagine your system is used by a company with 10,000 permanent employees. Retrieving all of those when you just wanted the names of a few temporary employees is going to take time. It also means that while you connect to a file or database, something else may not be able to.

Reading data from files or databases is also something that can have temporary errors, eg the network dropped for a moment or a file was locked, so doing so unnecessarily introduces more scope for your program going wrong.